test/MiniCodec.test.js

```
@@ -0,0 +1,64 @@
+        preferences: {
+            theme: 'dark',
+            notifications: true
+        }
+    }
+};
+
+const encoded = codec.encode('user', userData);
+const decoded = codec.decode('user', encoded);
+
+const dataMatches = JSON.stringify(userData) === JSON.stringify(decoded);
+console.assert(dataMatches, "Decoded data does not match the original user data. Original: " + JSON.stringify(userData) + ", Decoded: " + JSON.stringify(decoded));
```

Biome

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)